Source/WebCore/css/makeprop.pl

```diff
@@ -1074,30 +1035,48 @@ sub sortWithPrefixedPropertiesLast
 inline bool operator!=(const CSSPropertySettings& a, const CSSPropertySettings& b) { return !(a == b); }
 void add(Hasher&, const CSSPropertySettings&);
 
-bool isInternalCSSProperty(CSSPropertyID);
-bool isCSSPropertyExposed(CSSPropertyID, const Settings*);
-bool isCSSPropertyExposed(CSSPropertyID, const CSSPropertySettings*);
-ASCIILiteral getPropertyName(CSSPropertyID);
-const AtomString& getPropertyNameAtomString(CSSPropertyID id);
-String getPropertyNameString(CSSPropertyID id);
-String getJSPropertyName(CSSPropertyID);
-CSSPropertyID getRelatedPropertyId(CSSPropertyID id);
+constexpr bool isLonghand(CSSPropertyID);
+bool isInternal(CSSPropertyID);
+bool isExposed(CSSPropertyID, const Settings*);
+bool isExposed(CSSPropertyID, const CSSPropertySettings*);
+
+CSSPropertyID findCSSProperty(const char* characters, unsigned length);
+ASCIILiteral nameLiteral(CSSPropertyID);
+const AtomString& nameString(CSSPropertyID);
+String nameForIDL(CSSPropertyID);
+
+CSSPropertyID relatedProperty(CSSPropertyID);
+
+template<CSSPropertyID first, CSSPropertyID last> struct CSSPropertiesRange {
+    struct Iterator {
+        uint16_t index { static_cast<uint16_t>(first) };
+        constexpr CSSPropertyID operator*() const { return static_cast<CSSPropertyID>(index); }
+        constexpr Iterator& operator++() { ++index; return *this; }
+        constexpr bool operator!=(std::nullptr_t) const { return index <= static_cast<uint16_t>(last); }
+    };
+    static constexpr Iterator begin() { return { }; }
```

Comment:
Might be a good idea to add a `size()` function that returns `last - first` so that the sized container algorithms can be a bit more efficient.

Reply:
I like that idea. I just learned about the `std::ranges` concept `sized_range`.

But is there somewhere I would use that in this patch?

Reply:
It’s `last - first + 1`, because `last + 1 == end`.

Reply:
I had a sinking suspicion I would have an off by one there :).